buffer/src/main/java/io/netty/buffer/SimpleLeakAwareByteBuf.java

```diff
@@ -16,99 +16,161 @@
 
 package io.netty.buffer;
 
-import io.netty.util.ResourceLeak;
+import io.netty.util.ResourceLeakTracker;
+import io.netty.util.internal.ObjectUtil;
 
 import java.nio.ByteOrder;
 
-final class SimpleLeakAwareByteBuf extends WrappedByteBuf {
+class SimpleLeakAwareByteBuf extends WrappedByteBuf {
 
-    private final ResourceLeak leak;
+    // This is always the original ByteBuf whichs reference-count is responsible to signal when the ResourceLeakTracker
+    // should be closed. This means this always must be the ByteBuf that is used when calling LeakDetector.track(...).
+    private final ByteBuf trackedByteBuf;
```

Comment:
@Scottmitch see the ResourceLeakDetector why I now keep track of this one as well